### ISSUE TITLE
VSC cannot open notebook

### ISSUE THREAD

**Issue by halcwb** [dotnet/interactive#1775]
### Describe the bug
When trying to open an existing, previously functioning notebook, VSC reports:

> Cannot open resource with notebook editor type 'dotnet-interactive', please check if you have the right extension installed or enabled.

### Please complete the following:

**Which version of .NET Interactive are you using?** There are a few ways to find this out:

When I run `dotnet interactive --version`. I get not installed. Reinstalling the extension and restarting the terminal gives the same result.

- OS
    - [ ] Windows 11
    - [ ] Windows 10
    - [x] macOS
    - [ ] Linux (Please specify distro)
    - [ ] iOS
    - [ ] Android
 - Browser 
    - [x] Chrome
    - [ ] Edge
    - [ ] Firefox
    - [ ] Safari
 - Frontend
    - [ ] Jupyter Notebook
    - [ ] Jupyter Lab
    - [ ] nteract
    - [x] Visual Studio Code 
    - [ ] Visual Studio Code Insiders
    - [ ] Other (please specify)

**Comment by colombod** [comment]
What version of MacOs, vscode and extesnion do you have?

**Comment by halcwb** [comment]
.NET SDK (reflecting any global.json):
 Version:   5.0.402
 Commit:    e9d3381880

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  11.0
 OS Platform: Darwin
 RID:         osx.11.0-x64
 Base Path:   /usr/local/share/dotnet/sdk/5.0.402/

Host (useful for support):
  Version: 5.0.11
  Commit:  f431858f8b

.NET SDKs installed:
  2.1.101 [/usr/local/share/dotnet/sdk]
  2.1.302 [/usr/local/share/dotnet/sdk]
  2.1.402 [/usr/local/share/dotnet/sdk]
  2.1.505 [/usr/local/share/dotnet/sdk]
  2.1.700 [/usr/local/share/dotnet/sdk]
  2.2.301 [/usr/local/share/dotnet/sdk]
  3.0.100 [/usr/local/share/dotnet/sdk]
  3.1.100-preview3-014645 [/usr/local/share/dotnet/sdk]
  3.1.100 [/usr/local/share/dotnet/sdk]
  3.1.101 [/usr/local/share/dotnet/sdk]
  3.1.200 [/usr/local/share/dotnet/sdk]
  3.1.201 [/usr/local/share/dotnet/sdk]
  3.1.202 [/usr/local/share/dotnet/sdk]
  3.1.300 [/usr/local/share/dotnet/sdk]
  3.1.301 [/usr/local/share/dotnet/sdk]
  3.1.401 [/usr/local/share/dotnet/sdk]
  3.1.402 [/usr/local/share/dotnet/sdk]
  3.1.403 [/usr/local/share/dotnet/sdk]
  3.1.411 [/usr/local/share/dotnet/sdk]
  5.0.100 [/usr/local/share/dotnet/sdk]
  5.0.302 [/usr/local/share/dotnet/sdk]
  5.0.401 [/usr/local/share/dotnet/sdk]
  5.0.402 [/usr/local/share/dotnet/sdk]

VSCode uptodate

Extensions:
- Ionide for F#
- Bracket colorizer v2
- Jupiter notebooks 
- C#

**Comment by colombod** [comment]
@rebornix

**Comment by colombod** [comment]
Did you update vscode regurarly on your macos?

**Comment by halcwb** [comment]
I tend to keep it updated to the latest version whenever I use it.

**Comment by colombod** [comment]
Can you try to launch from command line with `--user-data-dir` pointing to a new empty folder and see if there is issues with the UserData folder state?

**Comment by halcwb** [comment]
You mean VSC or dotnet interactive?

**Comment by colombod** [comment]
visualstudio code

**Comment by halcwb** [comment]
So I did

```
open /Applications/Visual\ Studio\ Code.app/ --args --user-data-dir temp
```

Opens up VSC with the same result.

**Comment by colombod** [comment]
do you have a lot of setup? would you be up to test deleting the user data cache?

**Comment by halcwb** [comment]
Depends on what you mean with the data cache :-). I don't run a lot of extensions, if you mean that.

**Comment by colombod** [comment]
[Here](https://code.visualstudio.com/docs/getstarted/settings#_settings-file-locations) there are the file locations, I was able to get all working on macos with latest .NET5 and .NET6 sdks (6 is for insider only) and deleted the folder in the library completely.

@rebornix is looking at my settigns cache to figure out a much less aggressive way to address this issue

**Comment by halcwb** [comment]
Ok, so, essentially it is the ``settings.json`` you need to delete? Did you try just moving the file and the copy it back?

**Comment by halcwb** [comment]
The solution was very simple, just

```bash
dotnet tool install -g Microsoft.dotnet-interactive
```

did the trick. 

Don't know why or how this tool did get uninstalled though.

**Comment by halcwb** [comment]
An I have to reopen this issue again. Whenever I close VSC and reopen the project, VSC seems unable to find interactive. See screenshot:
<img width="1357" alt="2021-11-23_09-45-46" src="https://user-images.githubusercontent.com/683631/142993815-aebc6310-d3c0-4bf1-bc2c-96c167cb0294.png">

So, if I run `dotnet interactive --version`, the version is shown, so interactive is running.

**Comment by halcwb** [comment]
P.S. When I close the window and reopen the window, then I can use interactive again. But I need to do that each time after closing VSC and reopening VSC, automatically opening the last closed window resulting in the above problem.

**Comment by colombod** [comment]
Installing the global tool won't help to resolve this issue unfortunately. I'd rather try to see what si going on if you are ok with a screen share session.

**Comment by halcwb** [comment]
I think this is a versioning problem. I now have the same problem on my Windows machine. You cannot use the dotnet-interactive as the current version seems only compatible with the new 6.0 SDK. If you look at the dotnet-interactive versions it is not clear which version works with which SDK.

**Comment by colombod** [comment]
> I think this is a versioning problem. I now have the same problem on my Windows machine. You cannot use the dotnet-interactive as the current version seems only compatible with the new 6.0 SDK. If you look at the dotnet-interactive versions it is not clear which version works with which SDK.

.NET interactive requires .NET 6 from the december update, that is reported on teh extension page to
<img width="445" alt="image" src="https://user-images.githubusercontent.com/375556/146905500-d77f7295-e5e2-464a-a9e2-6cae9d58503a.png">

What versions of VSCode and .NET Intreactive a re installed?
Do you update vscode frequently?

**Comment by colombod** [comment]
Can you try a command added by VSCode to correct setting data format?
in the palette would be this one
<img width="248" alt="image" src="https://user-images.githubusercontent.com/375556/146906139-7fc5d102-a0eb-4ded-8a8b-d1a40a3b23f3.png">

**Comment by halcwb** [comment]
@colombod Thanks, you're right, my bad, didn't read the f* manual :-).

Installed the SDK and now everything works fine.

**Comment by colombod** [comment]
It happens to me a lot :( Thank you for your patience, I wish you the best holiday break there is, see you in 2022

Dr Diego Colombo PhD

**Comment by jonsequitur** [comment]
Can this be closed?

**Comment by iainsproat** [comment]
I experienced this same issue while using VSCode connected remotely to Windows Subsystem for Linux (WSL) running Ubuntu 20.04.

For anyone that experiences this issue in the future, these were my steps to resolve the problem:
1. installed dotnet interactive Notebooks extension for VSCode.
1. Tried running `.Net Interactive: Create new blank notebook` from the VSCode Command Palette (I experienced the reported issue at this point.)
1. opened the WSL terminal (either in VSCode terminal, or in the WSL app)
1. in the terminal, checked the installed dotnet SDK with `dotnet --info`; I only had .Net 3.1 installed.
1. installed dotnet 6.0 using the instructions here: https://docs.microsoft.com/en-us/dotnet/core/install/linux-ubuntu#install-the-sdk
1. verified that this installed correctly, `dotnet --info`
1. installed dotnet interactive: `dotnet tool install -g Microsoft.dotnet-interactive`
1. checked this installed correctly `dotnet-interactive --version`
1. closed and then re-opened VSCode; it then worked 😅 

Hope this helps someone!

**Comment by DavidZidar** [comment]
I just got hit with this issue. I'm on Windows 11, I have the latest VS Code version (updated just now), the latest .NET Interactive extension (also just updated), I have .NET 6 version 6.0.101 installed and I even installed the dotnet tool `dotnet tool install -g Microsoft.dotnet-interactive`, I also tried to clear the editor data cache as mentioned above but nothing seems to be helping.

If I try to create a new blank notebook I get the following error:
![image](https://user-images.githubusercontent.com/381720/152435538-7a02c105-cba4-41d7-bb64-ebd62ec7c1b4.png)

These are the only logs I could find:
```
Visual Studio Code (1.64.0, undefined, desktop)
Jupyter Extension Version: 2022.1.1001775990.
Python Extension not installed.
User belongs to experiment group 'jupyterTest'
User belongs to experiment group 'jupyterEnhancedDataViewer'
Info 23:01:20: ZMQ install verified.
```

What can I try next? Any more logs to check?
